### PR TITLE
Update model-card for DINOv2

### PR DESCRIPTION
**Pull request by shubham0204**
# What does this PR do?

This PR updates the model-card for the `dinov2` model, as described in #36979, in an attempt to standardize all model-cards.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@stevhliu

### CODE REVIEW

File: docs/source/en/model_doc/dinov2.md [hunk 1]
```diff
@@ -10,38 +10,51 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
-">
-<img alt="FlashAttention" src="https://img.shields.io/badge/%E2%9A%A1%EF%B8%8E%20FlashAttention-eae0c8?style=flat">
-<img alt="SDPA" src="https://img.shields.io/badge/SDPA-DE3412?style=flat&logo=pytorch&logoColor=white">
-</div>
+[DINOv2](https://huggingface.co/papers/2304.07193) is a vision foundation model that is used as a feature extractor for multiple downstream tasks like image classification and depth estimation. The training consists of a student and a teacher model that have the same [`Vision Transformer (ViT)`] architecture. The student model is optimized with gradient-descent to produce outputs similar to the teacher model. Embeddings of uncurated images are matched with embeddings of curated images through k-NN, resulting in a larger dataset for pretraining.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:25]:
```suggestion
[DINOv2](https://huggingface.co/papers/2304.07193) is a vision foundation model that uses [ViT](./vit) as a feature extractor for multiple downstream tasks like image classification and depth estimation. It focuses on stabilizing and accelerating training through techniques like a faster memory-efficient attention, sequence packing, improved stochastic depth, Fully Sharded Data Parallel (FSDP), and model distillation.
```

File: docs/source/en/model_doc/dinov2.md [hunk 2]
```diff
@@ -10,38 +10,51 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-## Overview
+You can find all the original DINOv2 checkpoints under the [Dinov2](https://huggingface.co/collections/facebook/dinov2-6526c98554b3d2576e071ce3) release.
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:27]:
```suggestion
You can find all the original DINOv2 checkpoints under the [Dinov2](https://huggingface.co/collections/facebook/dinov2-6526c98554b3d2576e071ce3) collection.
```

File: docs/source/en/model_doc/dinov2.md [hunk 3]
```diff
@@ -10,38 +10,51 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+<hfoptions id="usage">
+<hfoption id="Pipeline">
 
-This model was contributed by [nielsr](https://huggingface.co/nielsr).
-The original code can be found [here](https://github.com/facebookresearch/dinov2).
+```py
+import torch
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:38]: I think for both the `Pipeline` and `AutoModel` examples, it'd be more practical to show an image classification example.

We can also show a quantization example with the facebook/dinov2-giant checkpoint!

File: docs/source/en/model_doc/dinov2.md [hunk 4]
```diff
@@ -50,31 +63,14 @@ processor = AutoImageProcessor.from_pretrained('facebook/dinov2-base')
...
-
-<PipelineTag pipeline="image-classification"/>
+</hfoption>
 
-- [`Dinov2ForImageClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/image-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/image_classification.ipynb).
-- See also: [Image classification task guide](../tasks/image_classification)
+## Notes
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
+- [`DINOv2 with Registers`]
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:73]: Add indented code snippet of tracing under here.

```suggestion
- Use [torch.jit.trace](https://pytorch.org/docs/stable/generated/torch.jit.trace.html) to speedup inference. However, it will produce some mismatched elements. The difference between the original and traced model is 1e-4.

   <code>
```

**Review by stevhliu — COMMENTED** [review]
Nice work, thanks! I think an image classification example would be more impactful since that is something users can immediately and visually grasp versus an embedding.

File: docs/source/en/model_doc/dinov2.md [hunk 5]
```diff
@@ -10,34 +10,71 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+image = Image.open(requests.get(url, stream=True).raw)
+
+processor = AutoImageProcessor.from_pretrained('facebook/dinov2-base')
+model = AutoModel.from_pretrained('facebook/dinov2-base')
+
+inputs = processor(images=image, return_tensors="pt")
+image_embedding = model(**inputs)[0][0].mean(dim=0)
+```
+
+</hfoption>
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:69]:
```suggestion
</hfoption>
</hfoptions>
```

File: docs/source/en/model_doc/dinov2.md [hunk 6]
```diff
@@ -10,34 +10,71 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
 
-This model was contributed by [nielsr](https://huggingface.co/nielsr).
-The original code can be found [here](https://github.com/facebookresearch/dinov2).
+```py
+import torch
+from transformers import pipeline
+
+pipe = pipeline(
+    task="image-feature-extraction", 
+    model="facebook/dinov2-small"
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:43]:
```suggestion
    model="facebook/dinov2-small",
    torch_dtype=torch.float16,
    device=0
```

File: docs/source/en/model_doc/dinov2.md [hunk 7]
```diff
@@ -10,34 +10,71 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+import requests
+import torch
+from transformers import AutoImageProcessor, AutoModel
+from PIL import Image
+
+url = 'http://images.cocodataset.org/val2017/000000039769.jpg'
+image = Image.open(requests.get(url, stream=True).raw)
+
+processor = AutoImageProcessor.from_pretrained('facebook/dinov2-base')
+model = AutoModel.from_pretrained('facebook/dinov2-base')
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:63]:
```suggestion
model = AutoModel.from_pretrained("facebook/dinov2-base", torch_dtype=torch.float16, device_map="auto", attn_implementation="sdpa")
```

File: docs/source/en/model_doc/dinov2.md [hunk 8]
```diff
@@ -10,34 +10,71 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+
+inputs = processor(images=image, return_tensors="pt")
+image_embedding = model(**inputs)[0][0].mean(dim=0)
+```
+
+</hfoption>
+
+## Notes
+
+- [`DINOv2 with Registers`]
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:73]:
```suggestion
```

File: docs/source/en/model_doc/dinov2.md [hunk 9]
```diff
@@ -10,34 +10,71 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+
+</hfoption>
+
+## Notes
+
+- [`DINOv2 with Registers`] 
+
+- Use [torch.jit.trace](https://pytorch.org/docs/stable/generated/torch.jit.trace.html) to speedup inference. However, it will produce some mismatched elements. The difference between the original and traced model is 1e-4.
+
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/dinov2.md:77]: Indent this code block so its aligned under the list item here

File: docs/source/en/model_doc/dinov2.md (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/model_doc/dinov2.md:38]: Yeah we can do something like this with a finetuned checkpoint:

```py
import torch
from transformers import pipeline

pipe = pipeline(
    task="image-classification", 
    model="facebook/dinov2-small-imagenet1k-1-layer",
    torch_dtype=torch.float16,
    device=0
)

pipe("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg")
```

**Review by stevhliu — COMMENTED** [review]
Thanks for your changes, just a few more and then we're ready!

**Comment by shubham0204** [comment]
@stevhliu All requested changes have been made in the latest commit 👍🏻

File: docs/source/en/model_doc/dinov2.md (hunk 3 above)
Comment (stevhliu) [reply docs/source/en/model_doc/dinov2.md:38]: Thanks, I think you may have forgotten a the quantization example with the https://huggingface.co/facebook/dinov2-giant-imagenet1k-1-layer checkpoint though!

Comment (stevhliu) [reply docs/source/en/model_doc/dinov2.md:38]: This still needs to be addressed ☝️

**Comment by shubham0204** [comment]
@stevhliu I have added a quantization example as requested 👍🏻

**Review by stevhliu — APPROVED** [review]
Awesome, thanks again for your changes! 🤗

File: docs/source/en/model_doc/dinov2.md [hunk 10]
```diff
@@ -10,71 +10,134 @@ an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express o
...
+    # We have to force return_dict=False for tracing
+    model.config.return_dict = False
 
-- [`Dinov2ForImageClassification`] is supported by this [example script](https://github.com/huggingface/transformers/tree/main/examples/pytorch/image-classification) and [notebook](https://colab.research.google.com/github/huggingface/notebooks/blob/main/examples/image_classification.ipynb).
-- See also: [Image classification task guide](../tasks/image_classification)
+    with torch.no_grad():
+        traced_model = torch.jit.trace(model, [inputs.pixel_values])
+        traced_outputs = traced_model(inputs.pixel_values)
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we'll review it! The resource should ideally demonstrate something new instead of duplicating an existing resource.
```
Comment (qubvel) [inline docs/source/en/model_doc/dinov2.md:77]: Providing code snippets is great, however it would be super nice to keep resources within docs, they usually provide more in-depth view of the model, e.g. finetuning 🤗 

cc @stevhliu
